src/rest.cpp

```diff
@@ -269,6 +269,56 @@ static bool rest_block_notxdetails(HTTPRequest* req, const std::string& strURIPa
     return rest_block(req, strURIPart, false);
 }
 
+static bool rest_blockhash(HTTPRequest* req, const std::string& strURIPart)
+{
+    if (!CheckWarmup(req))
+        return false;
+    std::string strHeight;
+    const RetFormat rf = ParseDataFormat(strHeight, strURIPart);
+
+    int32_t nHeight;
+    if (!ParseInt32(strHeight, &nHeight))
+        return RESTERR(req, HTTP_BAD_REQUEST, "Parse error");
+
+    LOCK(cs_main);
+    if (nHeight < 0 || nHeight > chainActive.Height())
+        return RESTERR(req, HTTP_BAD_REQUEST, "Block height out of range: " + strHeight);
+
+    const CBlockIndex *pindex = chainActive[nHeight];
+
+    switch (rf) {
+    case RetFormat::BINARY: {
+        CDataStream ssGetBlockHashResponse(SER_NETWORK, PROTOCOL_VERSION);
+        ssGetBlockHashResponse << pindex->GetBlockHash();
```

Comment:
Could the call to `pindex->GetBlockHash()` be refactored out of the switch statement, both to reduce code repetition and to minimise the scope of where `cs_main` is held?